refactor(compiler): remove all source-level traces to tsc-wrapped

matsko opened the pull request.

mary-poppins commented:
You can preview e096395 at https://pr18966-e096395.ngbuilds.io/.

mary-poppins commented:
You can preview a8f62d9 at https://pr18966-a8f62d9.ngbuilds.io/.

mary-poppins commented:
You can preview 325ce60 at https://pr18966-325ce60.ngbuilds.io/.

mary-poppins commented:
You can preview 6cb4d97 at https://pr18966-6cb4d97.ngbuilds.io/.

mary-poppins commented:
You can preview e24c8bc at https://pr18966-e24c8bc.ngbuilds.io/.

mary-poppins commented:
You can preview 84c3201 at https://pr18966-84c3201.ngbuilds.io/.

mary-poppins commented:
You can preview 9432bba at https://pr18966-9432bba.ngbuilds.io/.

mary-poppins commented:
You can preview d956a01 at https://pr18966-d956a01.ngbuilds.io/.

mary-poppins commented:
You can preview 87322fa at https://pr18966-87322fa.ngbuilds.io/.

IgorMinar commented on packages/compiler-cli/test/test_support.ts:
```diff
@@ -25,4 +25,4 @@ export function makeTempDir(): string {
   fs.mkdirSync(dir);
   return dir;
-}
+}
```
keep blank line

IgorMinar reviewed (commented):
the commit message refers to the packages/tsc-wrapped, but that has been since moved to a separate commit - please update.

otherwise this looks good to me. @tbosch should approve this before merging it since he has the most context for this change.

mary-poppins commented:
You can preview 860a0e0 at https://pr18966-860a0e0.ngbuilds.io/.

tbosch commented on modules/tsconfig-ngc.json:
```diff
@@ -0,0 +1,11 @@
+{
+  "extends": "./tsconfig",
```
Could you move this into the other wip PR that I should take over (which also contains your start on `test.sh` and `build.sh`?

tbosch commented on packages/compiler-cli/src/transformers/lower_expressions.ts:
```diff
@@ -35,6 +35,17 @@ function toMap<T, K>(items: T[], select: (item: T) => K): Map<K, T> {
   return new Map(items.map<[K, T]>(i => [select(i), i]));
 }
 
+// copied from typescript core
```
remove this.

tbosch commented on packages/tsconfig-ngc.json:
```diff
@@ -0,0 +1,11 @@
+{
+  "extends": "./tsconfig",
```
move into the other wip PR

matsko committed:
refactor(compiler): remove all source-level traces to tsc-wrapped

- temporarily keeps the old sources under packages/tsc-wrapped
  until the build scripts are changed to use compiler-cli everywhere.
- removes the compiler options `disableTransformerPipeline` that was introduced
  in a previous beta of Angular 5, i.e. the transformer based compiler
  is now always enabled.

mary-poppins commented:
You can preview 69dab6c at https://pr18966-69dab6c.ngbuilds.io/.

yogeshgadge commented on packages/compiler-cli/src/metadata/bundle_index_host.ts:
```diff
@@ -0,0 +1,89 @@
+        return delegate.getSourceFile(fileName, languageVersion, onError);
+      };
+
+  newHost.writeFile =
```
I believe the writeFile is already defined on WebpackCompilerHost as a getter (not configurable). This is causing https://github.com/angular/angular-cli/issues/8473
`TypeError: Cannot set property writeFile of #<WebpackCompilerHost> which has only a getter`
